src/test/java/uk/gov/companieshouse/controller/PaymentControllerTest.java

```diff
@@ -69,6 +77,60 @@ public void getPaymentUIDataRequest_returnsNotFound_ifDissolutionDoesntExist() t
                 .andExpect(status().isNotFound());
     }
 
+    @Test
+    public void patchPaymentDataRequest_returnsNotFound_ifDissolutionDoesntExist() throws Exception {
+        final PaymentPatchRequest body = generatePaymentPatchRequest();
+
+        when(dissolutionService.get(COMPANY_NUMBER)).thenReturn(Optional.empty());
+
+        mockMvc
+                .perform(
+                        patch(PAYMENT_URI, COMPANY_NUMBER)
+                                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                                .content(asJsonString(body))
+                )
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void patchPaymentDataRequest_returnsBadRequest_ifDissolutionStatusIsNotPendingPayment() throws Exception {
+        final DissolutionGetResponse dissolutionGetResponse = generateDissolutionGetResponse();
+        dissolutionGetResponse.setApplicationStatus(ApplicationStatus.PENDING_APPROVAL);
+
+        final PaymentPatchRequest body = generatePaymentPatchRequest();
+
+        when(dissolutionService.get(COMPANY_NUMBER)).thenReturn(Optional.of(dissolutionGetResponse));
+
+        mockMvc
+                .perform(
+                        patch(PAYMENT_URI, COMPANY_NUMBER)
+                                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                                .content(asJsonString(body))
+                )
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void patchPaymentDataRequest_returnsOK_ifDissolutionExistsAndItsStatusIsPendingPayment() throws Exception {
```

Comment:
We are missing test coverage for what happens if a failed or cancelled payment request is received.

Reply:
What should happen if the payment is failed or cancelled? It's not outlined in the api definition.

Reply:
It just shouldn't call the service to update the payment details. It shouldn't throw an error because it is invoked by the Payment API and a status code would trigger an error. Maybe add a log to capture the status of the payment to accomodate for the lack of an error?